Cleanup core

sudden6 opened the pull request:
**WARNING: MAKE A BACKUP BEFORE TESTING**

I'm not yet entirely sure this doesn't crash the database and the settings file.

I intend to clean up all the low level data structures in qTox.

- [X] Build low level classes for Tox ID and Keys
- [X] Stop passing `QString`s, use the low level classes instead (mostly done in low level classes, I'm not quite sure if I should also do this for the UI classes)
~- [ ] Switch all internal data structures from using badly defined strings to well defined classes~ Will be in another PR

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4027)
<!-- Reviewable:end -->

sudden6 committed:
refactor(toxpk): add class representing a Tox Public Key

sudden6 committed:
feat(toxid): reduce passing Tox IDs around

reasons:
- most of the time we don't even know the Tox ID but only the Public Key
- use well defined objects instead of strings that could be anything

Diadlo commented:
Just IMO. If you have a checklist -- it may be a reason to split it on few PRs ;)

Diadlo commented:
Reviewed 3 of 3 files at r1, 8 of 39 files at r2.
Review status: 8 of 41 files reviewed at latest revision, 10 unresolved discussions.

---

*[src/friend.cpp, line 33 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzudlOQsyGla93R8n:-K_EzudmErFGuswMutfO:bvpu31f) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friend.cpp#L33)):*
> ```cpp
> #include "src/group.h"
> 
> Friend::Friend(uint32_t FriendId, const ToxKey &FriendPk)
> ```

1) `s/ToxKey &/ToxKey& /`
2) Variables must begin with a lowercase letter

---

*[src/friend.h, line 53 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-8Y9ofzJ874X9QTi:-K_F-8YAynS0hm7JPpAa:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friend.h#L53)):*
> ```objc
>     int getEventFlag() const;
> 
>     const ToxKey &getPublicKey() const;
> ```

& follow the type

---

*[src/friendlist.h, line 32 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-MoRUMsrAb6iHYbn:-K_F-MoRUMsrAb6iHYbo:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friendlist.h#L32)):*
> ```objc
> {
> public:
>     static Friend* addFriend(int friendId, const ToxKey &friendPk);
> ```

& follow the type

---

*[src/friendlist.h, line 34 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-NImX76IRxUW3f8R:-K_F-NImX76IRxUW3f8S:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friendlist.h#L34)):*
> ```objc
>     static Friend* addFriend(int friendId, const ToxKey &friendPk);
>     static Friend* findFriend(int friendId);
>     static Friend* findFriend(const ToxKey &friendPk);
> ```

& follow the type

---

*[src/group.h, line 61 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-UkuF7HiHfvKp4K9:-K_F-UkvaIDb6sx7ORQi:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/group.h#L61)):*
> ```objc
>     QString getName() const;
> 
>     QString resolveToxId(const ToxKey &id) const;
> ```

& follow the type

---

*[src/core/core.cpp, line 578 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F0Dzec-4hXFvX1Uke:-K_F0DzfqPsMpmTXojru:b88vw3y) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/core/core.cpp#L578)):*
> ```cpp
> void Core::requestFriendship(const ToxId& friendAddress, const QString& message)
> {
>     // TODO: add check for valid friendAddress
> ```

Why don't extract public key in local variable like it was with `userId`?

---

*[src/core/core.h, line 68 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F1DM72Q-Q5XpQHkA5:-K_F1DM80O8GDtFnoVKc:bfbfrbd) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/core/core.h#L68)):*
> ```objc
>     ToxKey getGroupPeerKey(int groupId, int peerId) const;
>     QList<QString> getGroupPeerNames(int groupId) const;
>     //QString getFriendAddress(uint32_t friendNumber) const;
> ```

Why not removed?

---

*[src/core/toxkey.cpp, line 48 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_Ez-zzL4MhT7rWlDOb:-K_Ez-zzL4MhT7rWlDOc:bcfupw2) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxkey.cpp#L48)):*
> ```cpp
>  * @brief Constructs a ToxKey from bytes.
>  * @param rawId The bytes to construct the ToxKey from.
>  * @param len Number of bytes to read. Must be exactly TOX_PUBLIC_KEY_SIZE, else the ToxKey will be empty.
> ```

What is the reason to specify length, if it always should have 1 value?)

---

*[src/core/toxkey.cpp, line 64 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzOH5jLTYkLz5jHrI:-K_EzOH5jLTYkLz5jHrJ:bh2l4cz) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxkey.cpp#L64)):*
> ```cpp
> /**
>  * @brief Compares the equality of the ToxKey.
>  * @param other ToxKey to compare.
> ```

Start sentence with capital letter, please :)

---

*[src/core/toxkey.cpp, line 74 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzUWk0Xz0VgVeGwKQ:-K_EzUWk0Xz0VgVeGwKR:b-ya81ng) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxkey.cpp#L74)):*
> ```cpp
> /**
>  * @brief Compares the inequality of the ToxKey.
>  * @param other ToxKey to compare.
> ```

Same as above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
I use by-commit review. So part of comments can be redundant. Sorry for that

sudden6 commented:
Review status: 0 of 40 files reviewed at latest revision, 10 unresolved discussions.

---

*[src/friend.cpp, line 33 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzudlOQsyGla93R8n:-K_FK_85Mnjq9lZXWXZP:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friend.cpp#L33)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

1) `s/ToxKey &/ToxKey& /`
2) Variables must begin with a lowercase letter
</blockquote></details>

Done.

---

*[src/core/core.cpp, line 578 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F0Dzec-4hXFvX1Uke:-K_FLtiaYKIb2v5c831e:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/core/core.cpp#L578)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Why don't extract public key in local variable like it was with `userId`?
</blockquote></details>

Done.

---

*[src/core/core.h, line 68 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F1DM72Q-Q5XpQHkA5:-K_FJvPNmiYEhrnjEtox:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/core/core.h#L68)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Why not removed?
</blockquote></details>

Done.

---

*[src/core/toxpk.cpp, line 64 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzOH5jLTYkLz5jHrI:-K_FKFawuLvtMg0kmTIs:b3tp0yn) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxpk.cpp#L64)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Start sentence with capital letter, please :)
</blockquote></details>

that's the name of the parameter ;)

---

*[src/core/toxpk.cpp, line 74 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzUWk0Xz0VgVeGwKQ:-K_FKM0LZEcIfW6L4ynu:bb7od0r) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxpk.cpp#L74)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

same

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Review status: 0 of 40 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[src/core/toxpk.cpp, line 64 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_EzOH5jLTYkLz5jHrI:-K_Fp3L5bY42Ml1mvKGo:bmk1dd) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxpk.cpp#L64)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

that's the name of the parameter ;)
</blockquote></details>

Ok.. Sorry :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 29 of 41 files at r2, 4 of 30 files at r3.
Review status: 14 of 40 files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[src/core/toxpk.h, line 1 at r3](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GQYqcxc-qO08852ii:-K_GQYqcxc-qO08852ij:b-tfpk7g) ([raw file](https://github.com/qtox/qtox/blob/22c6d39ceb4916a8a3ab55b1e1e8b9914cde9667/src/core/toxpk.h#L1)):*
> ```objc
> #ifndef ToxPk_H
> ```

Use uppercase

---

*[src/net/toxme.cpp, line 305 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GQx1hgRTeNLmZ4XK7:-K_GQx1iVs8UTS2MsPvh:b-as1gbn) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/net/toxme.cpp#L305)):*
> ```cpp
> Toxme::ExecCode Toxme::deleteAddress(QString server, ToxKey id)
> {
>     const QString payload{"{\"public_key\":\""+id.toString().left(64)+"\","
> ```

Seems, now `.left(64)` can be removed

---

*[src/persistence/profile.h, line 89 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GSNSY_gDNGgGm7D-v:-K_GSNSZTA6ctvVGRxLn:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/persistence/profile.h#L89)):*
> ```objc
> 
> private slots:
>     void loadDatabase(const ToxId &id);
> ```

& follow the type

---

*[src/persistence/settings.cpp, line 1432 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GSpBMr6Y4f-pnSy12:-K_GSpBMr6Y4f-pnSy13:bc9q7wx) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/persistence/settings.cpp#L1432)):*
> ```cpp
> }
> 
> QString Settings::getAutoAcceptDir(const ToxKey& id) const
> ```

& follow the type
Also, fix other occupation in this file

---

*[src/widget/widget.cpp, line 992 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GTaDZVOQnsUCA8u-6:-K_GTaD_ka8S3kO9WAdO:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.cpp#L992)):*
> ```cpp
> }
> 
> void Widget::addFriend(int friendId, const ToxKey &friendPk)
> ```

& follow the type

---

*[src/widget/widget.h, line 32 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GThAgRaHkSD6DqhDy:-K_GThAgRaHkSD6DqhDz:bhzjcbg) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.h#L32)):*
> ```objc
> 
> #include "src/core/corestructs.h"
> ```

Extra empty line

---

*[src/widget/widget.h, line 131 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GTlb82QUYEcO8DL6V:-K_GTlb82QUYEcO8DL6W:bhc778c) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.h#L131)):*
> ```objc
>     void setStatusMessage(const QString &statusMessage);
>     void addFriend(int friendId, const ToxKey &friendPk);
>     void addFriendFailed(const ToxKey &userId, const QString& errorInfo = QString());
> ```

& follow the type (2 lines above)

---

*[src/widget/form/addfriendform.h, line 23 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GU609voIo59LoEMCm:-K_GU609voIo59LoEMCn:be2bj30) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/addfriendform.h#L23)):*
> ```objc
> #define ADDFRIENDFORM_H
> 
> #include "src/core/toxid.h"
> ```

We have not information about it in `CONTRIBUTING.md`, but I see, that we often use forward declaration instead of include. ToxId in this file used only by reference, use can add forward declaration instead of include. 
But feel free to leave in until we don't specify it in standard
(If you plan to use forward declaration, change other occupations)

---

*[src/widget/form/addfriendform.h, line 80 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GUUpO3BQMtzH9ZpXg:-K_GUUpO3BQMtzH9ZpXh:bxemw8p) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/addfriendform.h#L80)):*
> ```objc
>     void retranslateAcceptButton(QPushButton* acceptButton);
>     void retranslateRejectButton(QPushButton* rejectButton);
>     void deleteFriendRequest(const ToxId &toxId);
> ```

& follow the type

---

*[src/widget/form/genericchatform.cpp, line 359 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GVVP2GFs9hkVeGRKS:-K_GVVP2GFs9hkVeGRKT:b-7d1aub) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/genericchatform.cpp#L359)):*
> ```cpp
>     {
>         msg = ChatMessage::createChatMessage(authorStr, message, ChatMessage::ACTION, authorIsActiveProfile);
>         // TODO: figure out if the line below can be removed
> ```

It ~can't. 6 lines below:
```
if ( (author == previousId) && (prevMsgDateTime.secsTo(QDateTime::currentDateTime()) < getChatLog()->repNameAfter) )
    msg->hideSender();
```

---

*[src/widget/form/genericchatform.cpp, line 472 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GVjHfNEaRoVUoiXYt:-K_GVjHfNEaRoVUoiXYu:b-ya81ng) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/genericchatform.cpp#L472)):*
> ```cpp
> void GenericChatForm::addSystemInfoMessage(const QString &message, ChatMessage::SystemMessageType type, const QDateTime &datetime)
> {
>     // TODO: figure out if the line below can be removed
> ```

Same as above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 26 of 30 files at r3.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[src/core/toxpk.cpp, line 48 at r1](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_Ez-zzL4MhT7rWlDOb:-K_WBekQ70WvpSRDD0oV:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/5e464a9be39d58e23475bbed6e71fa4df0c249e6/src/core/toxpk.cpp#L48)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

What is the reason to specify length, if it always should have 1 value?)
</blockquote></details>

Done.

---

*[src/core/toxpk.h, line 1 at r3](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GQYqcxc-qO08852ii:-K_WC3sJ7u9l_JfhqCL-:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/22c6d39ceb4916a8a3ab55b1e1e8b9914cde9667/src/core/toxpk.h#L1)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Use uppercase
</blockquote></details>

Done.

---

*[src/net/toxme.cpp, line 305 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GQx1hgRTeNLmZ4XK7:-K_WCIAlIJdyDkCT13Gb:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/net/toxme.cpp#L305)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Seems, now `.left(64)` can be removed
</blockquote></details>

Done.

---

*[src/persistence/profile.h, line 89 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GSNSY_gDNGgGm7D-v:-K_WCVNv2-m1W2hK60lE:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/persistence/profile.h#L89)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/persistence/settings.cpp, line 1432 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GSpBMr6Y4f-pnSy12:-K_WDUjnZCop5elsMGes:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/persistence/settings.cpp#L1432)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
Also, fix other occupation in this file
</blockquote></details>

Done.

---

*[src/widget/widget.cpp, line 992 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GTaDZVOQnsUCA8u-6:-K_WF7cdtM66uJWmFvuI:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.cpp#L992)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/widget/widget.h, line 32 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GThAgRaHkSD6DqhDy:-K_WF6wjWYBKmZSs-W9V:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.h#L32)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Extra empty line
</blockquote></details>

Done.

---

*[src/widget/widget.h, line 131 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GTlb82QUYEcO8DL6V:-K_WF94DJr_MXVaDBbl9:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/widget.h#L131)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type (2 lines above)
</blockquote></details>

Done.

---

*[src/widget/form/addfriendform.h, line 23 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GU609voIo59LoEMCm:-K_WFDQSSFGQVIQsryEb:b9qfga9) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/addfriendform.h#L23)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

We have not information about it in `CONTRIBUTING.md`, but I see, that we often use forward declaration instead of include. ToxId in this file used only by reference, use can add forward declaration instead of include. 
But feel free to leave in until we don't specify it in standard
(If you plan to use forward declaration, change other occupations)

</blockquote></details>

I don't plan to use forward declarations for now, because I don't fully understand their effects/side effects.

---

*[src/widget/form/addfriendform.h, line 80 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GUUpO3BQMtzH9ZpXg:-K_WFp2axe2vBP9sqfS_:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/addfriendform.h#L80)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/widget/form/genericchatform.cpp, line 359 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GVVP2GFs9hkVeGRKS:-K_WGLg1Oqc9TjXxSnHR:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/genericchatform.cpp#L359)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

It ~can't. 6 lines below:
```
if ( (author == previousId) && (prevMsgDateTime.secsTo(QDateTime::currentDateTime()) < getChatLog()->repNameAfter) )
    msg->hideSender();
```
</blockquote></details>

Done.

---

*[src/widget/form/genericchatform.cpp, line 472 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_GVjHfNEaRoVUoiXYt:-K_WGMW6sjg6owL2BeHG:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/widget/form/genericchatform.cpp#L472)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

zetok commented:
Reviewed 1 of 13 files at r4.
Review status: 28 of 40 files reviewed at latest revision, 22 unresolved discussions.

---

*[src/group.cpp, line 51 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZGafaxcaseVxWRrgJ:-K_ZGafbgTi7yUhnvdIf:b-mrd0ei) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/group.cpp#L51)):*
> ```cpp
> void Group::updatePeer(int peerId, QString name)
> {
>     ToxPk peerKey = Core::getInstance()->getGroupPeerKey(groupId, peerId);
> ```

`s/peerKey/peerPk/` ?

---

*[src/core/core.cpp, line 624 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZJcv3SiRAixF1owy7:-K_ZJcv4jKOwx8z3CHDY:bvz6cr7) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L624)):*
> ```cpp
>             Profile* profile = Nexus::getProfile();
>             if (profile->isHistoryEnabled())
>                 profile->getHistory()->addNewMessage(friendAddress.toString(), inviteStr, getSelfId().getPublicKey().toString(), QDateTime::currentDateTime(), true, QString());
> ```

Perhaps split on several lines?

---

*[src/core/core.cpp, line 625 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZJkZHGY0W4WQma3R6:-K_ZJkZIgki0ckmf1kbI:b-hz8fdb) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L625)):*
> ```cpp
>             if (profile->isHistoryEnabled())
>                 profile->getHistory()->addNewMessage(friendAddress.toString(), inviteStr, getSelfId().getPublicKey().toString(), QDateTime::currentDateTime(), true, QString());
>             // TODO: end
> ```

Style: add a newline between `// TODO` and branch of `if`

---

*[src/core/core.cpp, line 1118 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_YceHBWzqxPoQCruMY:-K_YceHBWzqxPoQCruMZ:bt3ldza) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L1118)):*
> ```cpp
>  * @brief Get the public key of a peer of a group
>  */
> ToxPk Core::getGroupPeerKey(int groupId, int peerId) const
> ```

Perhaps change `Key` to `Pk` ?

---

*[src/core/core.cpp, line 1450 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZKYUgemwsMvDpGv5X:-K_ZKYUhbGRlsyTULsr_:b-1d8eg0) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L1450)):*
> ```cpp
> void Core::setNospam(uint32_t nospam)
> {
>     // TODO: make this endian independent
> ```

`FIXME` would be better here

---

*[src/core/toxpk.h, line 19 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZGr64zPi573FP90eI:-K_ZGr64zPi573FP90eJ:bnz3ckn) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/toxpk.h#L19)):*
> ```objc
>     bool operator!=(const ToxPk& other) const;
>     QString toString() const;
>     QByteArray getKey() const;
> ```

Perhaps `getPkBytes` would be better?

Does C++ allow same function names for different functions when functions have different args/return type? If so, it could be just `getBytes`. Or is it a bad idea to have same function names when types differ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
Review status: 28 of 40 files reviewed at latest revision, 22 unresolved discussions.

---

*[src/friend.h, line 53 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-8Y9ofzJ874X9QTi:-K_aAaICQo9zHCe_OMGN:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friend.h#L53)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/friendlist.h, line 32 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-MoRUMsrAb6iHYbn:-K_aAbH2IWnDNUvIxKTe:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friendlist.h#L32)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/friendlist.h, line 34 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-NImX76IRxUW3f8R:-K_aAcQEs7tBuFjWq27t:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/friendlist.h#L34)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---

*[src/group.cpp, line 51 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZGafaxcaseVxWRrgJ:-K_a5RcY3VVYIOXM9dZN:b-8qbyn3) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/group.cpp#L51)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

`s/peerKey/peerPk/` ?
</blockquote></details>

this would create a name clash with the line below

---

*[src/core/core.cpp, line 624 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZJcv3SiRAixF1owy7:-K_a637g8h0-IFUT7kyQ:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L624)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

Perhaps split on several lines?
</blockquote></details>

Done.

---

*[src/core/core.cpp, line 625 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZJkZHGY0W4WQma3R6:-K_a64sPZGx3TBH6nGF5:b-rv8icd) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L625)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

Style: add a newline between `// TODO` and branch of `if`
</blockquote></details>

Done. I added the space afterwards though, because it marks the end of the block

---

*[src/core/core.cpp, line 1118 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_YceHBWzqxPoQCruMY:-K_a6U5UiCuJ7Sc7_edU:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L1118)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

Perhaps change `Key` to `Pk` ?
</blockquote></details>

Done.

---

*[src/core/core.cpp, line 1450 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZKYUgemwsMvDpGv5X:-K_aAFKvIihx7KzZrGOQ:b-xvqw6f) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/core.cpp#L1450)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

`FIXME` would be better here
</blockquote></details>

actually the implementation was wrong, I fixed it right away

---

*[src/core/toxpk.h, line 19 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_ZGr64zPi573FP90eI:-K_aARLHnyQ6qpFvkXuL:bckz218) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/toxpk.h#L19)):*
<details><summary><i>Previously, zetok wrote…</i></summary><blockquote>

Perhaps `getPkBytes` would be better?

Does C++ allow same function names for different functions when functions have different args/return type? If so, it could be just `getBytes`. Or is it a bad idea to have same function names when types differ?
</blockquote></details>

C++ allows this only for different function arguments, but not different return types.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
Review status: 25 of 40 files reviewed at latest revision, 22 unresolved discussions.

---

*[src/group.h, line 61 at r2](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_F-UkuF7HiHfvKp4K9:-K_aAtCre-WgwE25MlXW:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/b8d9be7a62720d070d1e2dd067c585895d11fbf2/src/group.h#L61)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

& follow the type
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 2 of 30 files at r3, 10 of 13 files at r4, 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[src/core/toxpk.cpp, line 47 at r5](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_aURjtuMdzTIeYlKl-:-K_aURjtuMdzTIeYlKl0:bkuk0lv) ([raw file](https://github.com/qtox/qtox/blob/dbc75574a47905a962045692c697024611955204/src/core/toxpk.cpp#L47)):*
> ```cpp
> /**
>  * @brief Constructs a ToxKey from bytes.
>  * @param rawId The bytes to construct the ToxKey from, will read exactly
> ```

`s/ToxKey/ToxPk/`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

sudden6 commented:
Review status: 39 of 40 files reviewed at latest revision, 7 unresolved discussions.

---

*[src/core/toxpk.cpp, line 47 at r5](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_aURjtuMdzTIeYlKl-:-K_aYFaJiJqnc6IyvSoo:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/dbc75574a47905a962045692c697024611955204/src/core/toxpk.cpp#L47)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

`s/ToxKey/ToxPk/`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 1 of 30 files at r3, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[src/core/toxpk.cpp, line 46 at r4](https://reviewable.io:443/reviews/qtox/qtox/4027#-K_abZurYmrS7QVrjlLS:-K_abZurYmrS7QVrjlLT:bbdde0v) ([raw file](https://github.com/qtox/qtox/blob/0ff4a31952dfae7fe58dea011fa058b7a3d54ba2/src/core/toxpk.cpp#L46)):*
> ```cpp
> 
> /**
>  * @brief Constructs a ToxKey from bytes.
> ```

And here :)
`s/ToxKey/ToxPk/`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
Reviewed 1 of 30 files at r3, 1 of 13 files at r4, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027)*
<!-- Sent from Reviewable.io -->

Diadlo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4027#-:-K_acriXKS1Yjy3nldvi:bnfp4nl)*
<!-- Sent from Reviewable.io -->
